presto-main/src/test/java/io/prestosql/sql/planner/TestDomainTranslator.java

```diff
@@ -1582,6 +1582,39 @@ public void testLikePredicate()
         assertUnsupportedPredicate(not(like(C_VARCHAR, stringLiteral("abc\\_def"))));
     }
 
+    @Test
+    public void testStartsWithFunction()
+    {
+        Type varcharType = createUnboundedVarcharType();
+
+        // constant
+        testSimpleComparison(
+                startsWith(C_VARCHAR, stringLiteral("abc")),
+                C_VARCHAR,
+                startsWith(C_VARCHAR, stringLiteral("abc")),
+                Domain.create(ValueSet.ofRanges(Range.range(varcharType, utf8Slice("abc"), true, utf8Slice("abd"), false)), false));
+
+        testSimpleComparison(
+                startsWith(C_VARCHAR, stringLiteral("_abc")),
+                C_VARCHAR,
+                startsWith(C_VARCHAR, stringLiteral("_abc")),
+                Domain.create(ValueSet.ofRanges(Range.range(varcharType, utf8Slice("_abc"), true, utf8Slice("_abd"), false)), false));
+
+        // empty
+        assertUnsupportedPredicate(startsWith(C_VARCHAR, stringLiteral("")));
+
+        // non-ASCII
+        testSimpleComparison(
```

Comment:
Could you add test for `complement` case?

Reply:
Added.